File: binary-tree-right-side-view/binary-tree-right-side-view.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public: vector<int>ans;
    vector<int> rightSideView(TreeNode* root) {
        if(root == NULL)
        {
            return ans;
        }
        queue<TreeNode*> q;
        q.push(root);
        while(q.size()>0)
        {
            int size = q.size();
            while(size>0)
            {
                if(size==1)
                {
                    ans.push_back(q.front()->val);
                }
                TreeNode* node = q.front();
                q.pop();
                if(node->left!=nullptr)
                {
                    q.push(node->left);
                }
                if(node->right!=nullptr)
                {
                    q.push(node->right);
                }
                size--;
            }
        }
        return ans;
    }
};

File: binary-tree-right-side-view/binary-tree-right-side-view.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

from collections import deque
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        lst = []
        if not root:
            return lst;
        q = deque();
        q.append(root);
        while len(q)>0:
            size = len(q)
            while size>0:
                if size == 1:
                    lst.append(q[0].val);
                node = q[0];
                q.popleft();
                if node.left:
                    q.append(node.left);
                if node.right:
                    q.append(node.right)
                size-=1
        return lst;